aspnetcore/grpc/index.md

```diff
@@ -0,0 +1,185 @@
+
+### Configure `Startup`
+
+gRPC is enabled in the `Startup.ConfigureServices` method through the `AddGrpc` method:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Startup.cs?highlight=18)]
+
+Each gRPC service is added to the ASP.NET Core routing pipeline through the `MapGrpc` method in `Startup.Configure`:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Startup.cs?highlight=29-32)]
+
+Since ASP.NET Core middlewares and features share the routing pipeline, an application can be configured to serve additional request handlers such as MVC controllers in parallel with the configured gRPC services.
+
+## Integration with ASP.NET Core APIs
+
+gRPC services have full access to the ASP.NET Core features such as Dependency Injection (DI) and Logging. For example, the service implementation can resolve a logger from the DI container via the constructor:
+
+```csharp
+public class GreeterService : Greeter.GreeterBase
+{
+    public GreeterService(ILogger<GreeterService> logger)
+    {
+    }
+}
+```
+
+By default, the service implementation can resolve services with Singleton and Scoped lifetimes.
+
+### Resolve HttpContext in gRPC methods
+
+The gRPC API provides access to some underlying data of the HTTP/2 message such as the method, host, header and trailers through the `ServerCallContext` argument passed to each gRPC method:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Services/GreeterService.cs?highlight=12)]
+
+While this may be sufficient for many purposes, to ensure full access to ASP.NET Core APIs, the `HttpContext` representing the underlying HTTP/2 message can be accessed through the `GetHttpContext` extension method:
+
+```csharp
+public class GreeterService : Greeter.GreeterBase
+{
+    public override Task<HelloReply> SayHello(HelloRequest request, ServerCallContext context)
+    {
+        HttpContext httpContext = context.GetHttpContext();
+
+        return Task.FromResult(new HelloReply
+        {
+            Message = "Using https: " + httpContext.Request.IsHttps
+        });
+    }
+}
+```
+
+## gRPC service implementation lifetime
+
+By default, services will be created with a Scoped lifetime as defined in the [Dependency Injection](xref:fundamentals/dependency-injection) document. This allows the service implementation to resolve services with Scoped lifetimes, such as DBContext, from the DI container through constructor injection.
+
+> [!WARNING]
+> The Scoped lifetime of service implementation in ASP.NET Core is a behavioral difference from grpc C Core. Since a new instance of the service implementation is constructed for each request, it's no longer possible to share state between requests via instance members on the implementation type. Instead, the expectation is to store shared states in a Singleton service in the DI container and resolve it in the constructor of the gRPC service implementation.
```

Comment:
```suggestion
> The Scoped lifetime of service implementation in ASP.NET Core is a behavioral difference from gRPC C Core. Since a new instance of the service implementation is constructed for each request, it's no longer possible to share state between requests via instance members on the implementation type. Instead, the expectation is to store shared states in a Singleton service in the DI container and resolve it in the constructor of the gRPC service implementation.
```